wp-content/plugins/core/src/Blocks/Types/Icon_Grid/Icon_Grid.php

```diff
@@ -0,0 +1,129 @@
+
+	public const ICONS            = 'icons';
+	public const ICON_IMAGE       = 'image';
+	public const ICON_TITLE       = 'icon_title';
+	public const ICON_DESCRIPTION = 'icon_description';
+	public const ICON_LINK        = 'link';
+
+	public function add_block() {
+		$this->set_block( new Block( self::NAME, [
+			'title'       => __( 'Icon Grid', 'tribe' ),
+			'description' => __( 'A grid layout block with icon selections', 'tribe' ), // TODO: describe the block
+			'icon'        => '<svg enable-background="new 0 0 146.3 106.3" version="1.1" viewBox="0 0 146.3 106.3" xml:space="preserve" xmlns="http://www.w3.org/2000/svg"><style type="text/css">.st0{fill:#16D690;}.st1{fill:#21A6CB;}.st2{fill:#008F8F;}</style><polygon class="st0" points="145.2 106.3 72.6 42.3 26.5 1.2 0 106.3"/><polygon class="st1" points="145.2 106.3 0 106.3 72.6 42.3 118.6 1.2"/><polygon class="st2" points="72.6 42.3 145.2 106.3 0 106.3"/></svg>', // TODO: set SVG icon
+			'keywords'    => [ __( 'icon', 'grid', 'tribe' ) ], // TODO: select appropriate keywords
+			'category'    => 'common', // core categories: common, formatting, layout, widgets, embed
+			'supports'    => [
+				'align'  => false,
+				'anchor' => true,
+				'html'   => false,
+			],
+		] ) );
+	}
+
+	/**
+	 * Register Fields for block
+	 */
+	public function add_fields() {
+		//==========================================
+		// Content Fields
+		//==========================================
+		$this->add_section( new Field_Section( self::SECTION_CONTENT, __( 'Content', 'tribe' ), 'accordion' ) )
+			->add_field( new Field( self::NAME . '_' . self::TITLE, [
+				'label' => __( 'Title', 'tribe' ),
+				'name'  => self::TITLE,
+				'type'  => 'text',
+				] )
+			)->add_field( new Field( self::NAME . '_' . self::LEADIN, [
+					'label' => __( 'Lead in', 'tribe' ),
+					'name'  => self::LEADIN,
+					'type'  => 'text',
+				] )
+			)->add_field( new Field( self::NAME . '_' . self::DESCRIPTION, [
+					'label' => __( 'Description', 'tribe' ),
+					'name'  => self::DESCRIPTION,
+					'type'  => 'wysiwyg',
+				] )
+			)->add_field( new Field( self::NAME . '_' . self::CTA, [
+					'label' => __( 'Call to Action', 'tribe' ),
+					'name'  => self::CTA,
+					'type'  => 'link',
+				] )
+			)->add_field(
+				$this->get_icon_section()
+			);
+
+		//==========================================
+		// Setting Fields
+		//==========================================
+		$this->add_section( new Field_Section( self::SECTION_SETTINGS, __( 'Settings', 'tribe' ), 'accordion' ) );
+	}
+
+	/**
+	 * @return Repeater
+	 */
+	protected function get_icon_section() {
+		$group = new Repeater( self::NAME . '_' . self::ICONS );
+		$group->set_attributes( [
+			'label'        => __( 'Icon Section', 'tribe' ),
+			'name'         => self::ICONS,
+			'layout'       => 'block',
+			'min'          => 0,
+			'max'          => 12,
+			'button_label' => 'Add Icon Section',
+		] );
+
+		$icon_image = new Field( self::ICON_IMAGE, [
```

Comment:
You don't have to make any of these changes but we've made some improvements to registering fields that are kind of handy and nice for code clutter.

```
$group = new Repeater( self::NAME . '_' . self::ICONS, [
	'label'        => __( 'Icon Section', 'tribe' ),
	'name'         => self::ICONS,
	'layout'       => 'block',
	'min'          => 0,
	'max'          => 12,
	'button_label' => 'Add Icon Section',
] );

$group->add_field(
    new Field( self::ICON_IMAGE, [
		'label'         => __( 'Icon Image', 'tribe' ),
		'name'          => self::ICON_IMAGE,
		'type'          => 'image',
		'return_format' => 'id',
		'preview_size'  => 'medium',
		'instructions'  => __( 'Recommended image size: 100px wide with any aspect ratio.', 'tribe' ),
	] )
 )->add_field(
    new Field( self::ICON_TITLE, [
		'label' => __( 'Icon Title', 'tribe' ),
		'name'  => self::ICON_TITLE,
		'type'  => 'text',
	] )
)->add_field(
    new Field( self::ICON_DESCRIPTION, [
		'label' => __( 'Icon Description', 'tribe' ),
		'name'  => self::ICON_DESCRIPTION,
		'type'  => 'wysiwyg',
     ] )
)->add_field(
    new Field( self::ICON_LINK, [
		'label' => __( 'Icon Section Link', 'tribe' ),
		'name'  => self::ICON_LINK,
		'type'  => 'link',
    ] )
);

```
We aren't very consistent here _yet_ so again, no change needed but FYI these are available.